### ISSUE TITLE
Assertion '_toolbar == null' failure in TextField double-tap

### ISSUE THREAD

**Issue by cvolzke4** [flutter/flutter#33740]
## Steps to Reproduce
We received this error stack trace from a crash report.

## Logs
```
Thread 0 main thread_AssertionError: 'package:flutter/src/widgets/text_selection.dart': Failed assertion: line 357 pos 12: '_toolbar == null': is not true.
at _AssertionError._doThrowNew	(errors_patch.dart:40 )
at _AssertionError._throwNew	(errors_patch.dart:36 )
at TextSelectionOverlay.showToolbar	(text_selection.dart:357 )
at EditableTextState.showToolbar	(editable_text.dart:1393 )
at _TextFieldState._handleDoubleTapDown	(text_field.dart:800 )
at _TextSelectionGestureDetectorState._handleTapDown	(text_selection.dart:821 )
at TapGestureRecognizer._checkDown.<anonymous closure>	(tap.dart:328 )
at GestureRecognizer.invokeCallback	(recognizer.dart:181 )
at TapGestureRecognizer._checkDown	(tap.dart:328 )
at TapGestureRecognizer.didExceedDeadlineWithEvent	(tap.dart:292 )
at PrimaryPointerGestureRecognizer.addAllowedPointer.<anonymous closure>	(recognizer.dart:429 )
at _rootRun	(zone.dart:1120 )
at _CustomZone.run	(zone.dart:1021 )
at _CustomZone.runGuarded	(zone.dart:923 )
at _CustomZone.bindCallbackGuarded.<anonymous closure>	(zone.dart:963 )
at _rootRun	(zone.dart:1124 )
at _CustomZone.run	(zone.dart:1021 )
at _CustomZone.bindCallback.<anonymous closure>	(zone.dart:947 )
at Timer._createTimer.<anonymous closure>	(timer_patch.dart:21 )
at _Timer._runTimers	(timer_impl.dart:382 )
at _Timer._handleMessage	(timer_impl.dart:416 )
at _RawReceivePortImpl._handleMessage	(isolate_patch.dart:171 )
```

**Comment by HansMuller** [comment]
@ocavue - to track this down it would help if you'd provide the output of `flutter doctor` and a small self-contained app that demos the problem.

**Comment by justinmc** [comment]
I actually was able to reproduce this.  Just tap like crazy on a TextField and eventually it will throw this error.

This is likely a regression caused by https://github.com/flutter/flutter/pull/31852

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: Scaffold(
        appBar: AppBar(
          title: Text('Test'),
        ),
        body: Center(
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            mainAxisAlignment: MainAxisAlignment.start,
            children: <Widget>[
              Container(height: 246),
              TextField(),
            ],
          ),
        ),
      ),
    );
  }
}
```

**Comment by ocavue** [comment]
> @ocavue - to track this down it would ...

@HansMuller you seem to @ the wrong person. 😂

**Comment by HansMuller** [comment]
@ocavue  - sorry about that!

**Comment by justinmc** [comment]
@cvolzke4 This should be fixed wit the referenced PR above.  You'll need to be on the [master channel](https://github.com/flutter/flutter/wiki/Flutter-build-release-channels) now to get this fix, or using a version greater than 1.7.0.
